packages/dataclass-editor/src/detail/dataclass/entity/EntityClassDatabaseTable.tsx

```diff
@@ -0,0 +1,31 @@
+import { BasicField, Collapsible, CollapsibleContent, CollapsibleTrigger, Flex, Input } from '@axonivy/ui-components';
+import { useEntityClass } from '../../../context/AppContext';
+import type { DataClassEntity } from '../../../data/dataclass';
+
+export const useEntityProperty = () => {
+  const { entityClass, setEntityClass } = useEntityClass();
+  const setProperty = <EKey extends keyof DataClassEntity>(key: EKey, value: DataClassEntity[EKey]) => {
+    const newEntityClass = structuredClone(entityClass);
+    newEntityClass.entity[key] = value;
+    setEntityClass(newEntityClass);
+  };
+  return setProperty;
+};
+
+export const EntityClassDatabaseTable = () => {
+  const { entityClass } = useEntityClass();
+  const setProperty = useEntityProperty();
```

Comment:
why not deliver the entity via the hook (you already have it there).


```suggestion
export const useEntityProperty = () => {
  const { entityClass, setEntityClass } = useEntityClass();
  const setProperty = <EKey extends keyof DataClassEntity>(key: EKey, value: DataClassEntity[EKey]) => {
    const newEntityClass = structuredClone(entityClass);
    newEntityClass.entity[key] = value;
    setEntityClass(newEntityClass);
  };
  return {entityClass, setProperty};
};

export const EntityClassDatabaseTable = () => {
  const { entityClass, setProperty} = useEntityProperty();
```